openverse_catalog/dags/providers/provider_api_scripts/science_museum.py

```diff
@@ -45,198 +27,202 @@
-
-
-def _get_license_version(source):
-    license_version = None
-    if source:
-        legal = source.get("legal")
-        if legal:
-            rights = legal.get("rights")
-            if type(rights) == list:
-                license_version = rights[0].get("usage_terms")
-    return license_version
-
-
-def _get_metadata(obj_attr):
-    metadata = {}
-    identifier = obj_attr.get("identifier")
-    if type(identifier) == list:
-        metadata["accession number"] = identifier[0].get("value")
-    name = obj_attr.get("name")
-    if type(name) == list:
-        metadata["name"] = name[0].get("value")
-    category = obj_attr.get("categories")
-    if type(category) == list:
-        metadata["category"] = category[0].get("value")
-    creditline = obj_attr.get("legal")
-    if type(creditline) == dict:
-        metadata["creditline"] = creditline.get("credit_line")
-    description = obj_attr.get("description")
-    if type(description) == list:
-        metadata["description"] = description[0].get("value")
-    return metadata
+                size[dim.get("dimension")] = (
+                    dim.get("value") if dim.get("units") == "pixels" else None
+                )
+        return size.get("height"), size.get("width")
+
+    @staticmethod
+    def _get_image_info(
+        processed: Dict,
+    ) -> Tuple[Optional[str], Optional[int], Optional[int], Optional[str]]:
+        height, width, filetype = None, None, None
+        image_data = processed.get("large")
+        if image_data is None:
+            image_data = processed.get("medium", {})
+
+        image_url = ScienceMuseumDataIngester.check_url(image_data.get("location"))
+        if image_url:
+            filetype = image_data.get("format")
+            height, width = ScienceMuseumDataIngester._get_dimensions(image_data)
+        return image_url, height, width, filetype
+
+    @staticmethod
+    def _get_first_list_value(key: str, attributes: Dict) -> str | None:
+        val = attributes.get(key)
+        if isinstance(val, list):
+            return val[0].get("value")
+        return None
+
+    @staticmethod
+    def _get_metadata(attributes):
+        metadata = {}
+        for item in [
+            ("identifier", "accession number"),
+            ("name", "name"),
+            ("categories", "category"),
+            ("description", "description"),
+        ]:
+            attr_key, metadata_key = item
+            val = ScienceMuseumDataIngester._get_first_list_value(attr_key, attributes)
+            if val is not None:
+                metadata[metadata_key] = val
+
+        creditline = attributes.get("legal")
+        if isinstance(creditline, dict):
+            line = creditline.get("credit_line")
+            if line is not None:
+                metadata["creditline"] = line
+
+        return metadata
+
+    @staticmethod
+    def _get_license(image_data):
+        rights = image_data.get("source", {}).get("legal", {}).get("rights")
```

Comment:
It seems like we're doing this a lot across various scripts, I wonder if something like [`deep-dict`](https://pypi.org/project/dict-deep/) or similar for these cases (or if it'd be significantly less performant to do so) 🤔 Definitely doesn't need to be changed here, just floating the idea!

Reply:
Oop, actually [`glom`](https://github.com/mahmoud/glom) might be better and looks more actively developed!

Reply:
I opened #625 to track this